src/DIPS.Xamarin.UI/Controls/FloatingActionMenu/FloatingActionMenu.xaml.cs

```diff
@@ -0,0 +1,307 @@
+                    AddButtonsToRelative(relativeLayout);
+                    break;
+                case AbsoluteLayout absoluteLayout:
+                    AddButtonsToAbsolute(absoluteLayout);
+                    break;
+                default:
+                    return;
+            }
+
+            m_overLay.GestureRecognizers.Add(m_closeMenuRecognizer);
+
+            m_parent = layout;
+            m_yTranslate = Size + m_spaceBetween;
+        }
+
+        private void AddButtonsToAbsolute(AbsoluteLayout parent)
+        {
+            parent.Children.Add(m_overLay, new Rectangle(0, 0, 1, 1), AbsoluteLayoutFlags.SizeProportional);
+            parent.Children.Add(
+                ExpandButton,
+                new Rectangle(XConstraint, YConstraint, AbsoluteLayout.AutoSize, AbsoluteLayout.AutoSize),
+                AbsoluteLayoutFlags.PositionProportional);
+            foreach (var child in Children)
+            {
+                child.FloatingActionMenuParent = this;
+                child.Size = Size;
+                parent.Children.Add(
+                    child,
+                    new Rectangle(XConstraint, YConstraint, AbsoluteLayout.AutoSize, AbsoluteLayout.AutoSize),
+                    AbsoluteLayoutFlags.PositionProportional);
+            }
+        }
+
+        private void AddButtonsToRelative(RelativeLayout parent)
+        {
+            parent.Children.Add(
+                m_overLay,
+                Constraint.RelativeToParent(p => p.X),
+                Constraint.RelativeToParent(p => p.Y),
+                Constraint.RelativeToParent(p => p.Width),
+                Constraint.RelativeToParent(p => p.Height));
+            parent.Children.Add(
+                ExpandButton,
+                Constraint.RelativeToParent(p => p.Width * XConstraint),
+                Constraint.RelativeToParent(p => p.Height * YConstraint));
+            foreach (var child in Children)
+            {
+                child.FloatingActionMenuParent = this;
+                child.Size = Size;
+                parent.Children.Add(
+                    child,
+                    Constraint.RelativeToParent(p => p.Width * XConstraint),
+                    Constraint.RelativeToParent(p => p.Height * YConstraint));
+            }
+        }
+
+        private void AdjustXPositions()
+        {
+            foreach (var child in Children)
+                if (m_parent is AbsoluteLayout)
+                {
+                    AbsoluteLayout.SetLayoutFlags(child, AbsoluteLayoutFlags.None);
+                    AbsoluteLayout.SetLayoutBounds(
+                        child,
+                        new Rectangle(ExpandButton.X + Size - child.Width, ExpandButton.Y, child.Width, child.Height));
+                }
+                else if (m_parent is RelativeLayout)
+                {
+                    RelativeLayout.SetXConstraint(child, Constraint.Constant(ExpandButton.X + Size - child.Width));
```

Comment:
Expandbutton is placed on the contentView itself? Did you test this working?

Reply:
Unsure what you mean?

Reply:
I just wonder if you verified all regarding RelativeLayout

Reply:
Yes, its working for both Relative and Absolute, and modality